src/user_database.ts

```diff
@@ -11,11 +11,13 @@ export interface UserDatabase {
   getName(): UserDatabaseName;
 
   // Run transactionFunction as a database transaction with a given config and arguments.
-  transaction<T extends any[], R>(transactionFunction: UserDatabaseTransaction<T, R>, config: TransactionConfig, ...args: T): Promise<R>;
+  transaction<R, T extends unknown[]>(transactionFunction: UserDatabaseTransaction<R, T>, config: TransactionConfig, ...args: T): Promise<R>;
```

Comment:
Why do we need to change the order of `T` and `R`?

My understanding was that `R` stood for "Response"

Reply:
Response is always in the template, T is not.  For consistency, R first.